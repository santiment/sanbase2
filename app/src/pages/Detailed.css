.detailed-head {
  overflow: hidden;
  margin-top: 32px;
  height: 84px;
  margin-right: 16px;
}

.detailed-name h1 {
  margin-top: 0;
}

.detailed-name {
  float: left;
  width: 50%;
}

.detailed-name p {
  font-size: 1.25rem;
  font-weight: 500;
  line-height: 1.1;
  color: "#3D4051";
  margin-bottom: 0.5rem;
  margin-left: 52px;
}

.detailed-name img {
  display: inline-block;
  vertical-align: middle;
}

.detailed-buttons {
  float: right;
  width: 50%;
  height: 100%;
  position: relative;
}

.detailed-buttons a.add-to-dashboard {
  border: none;
  border-radius: 4px;
  padding: 8px 11px;
  text-transform: uppercase;
  letter-spacing: 1px;
  background-color: #009663;
  color: #fff;
  font-size: 12px;
  font-weight: 400;
  opacity: 0.7;
  position: absolute;
  right: 0;
  bottom: 0;
}

.detailed-buttons a.add-to-dashboard:hover {
  opacity: 1;
  background-color: #025aa5;
  border-color: #01549b;
}

.detailed .nav {
  border-bottom: 1px solid #ccc;
  display: flex;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
}

.detailed .nav li {
  display: list-item;
  text-align: -webkit-match-parent;
  margin-bottom: -1px;
}

.detailed .nav li .nav-link {
  border: 1px solid;
  border-color: #ccc rgba(0, 0, 0, 0) #ccc #ccc;
  background-color: #f2f2f2;
  color: #3d4051;
  display: block;
}

.detailed .main-chart .nav li {
  width: 50%;
}

.detailed .activity-panel .nav .nav-link {
  font-size: 1.25rem;
  font-weight: 500;
  color: "#3D4051";
  padding: 1px 8px;
}

.detailed .main-chart .nav .nav-link {
  font-size: 24px;
  font-weight: 300;
  color: #171717;
  padding: 4px 15px;
}

.detailed .nav li:last-child .nav-link {
  border-color: #ccc;
}

.main-chart .diff {
  font-size: 70%;
}

.main-chart .diff.up {
  color: "#309d81";
}

.main-chart .diff.down {
  color: "#ca2d2d";
}

.main-chart .content div {
  display: none;
}

.detailed .activity-panel .nav li {
  width: 20%;
}

.detailed .nav li.active .nav-link {
  background-color: #fff;
  border-color: #ccc rgba(0, 0, 0, 0) #fff #ccc;
}

.detailed .nav li.active:last-child .nav-link {
  background-color: #fff;
  border-color: #ccc #ccc #fff #ccc;
}

.detailed .analysis .panel {
  width: 33%;
  float: left;
  margin: 16px 0.33333% 16px 0;
}

.detailed .information .panel {
  width: 49.75%;
  float: left;
  margin: 16px 0.5% 16px 0;
}

.detailed .information .panel:last-child {
  margin: 16px 0% 16px 0;
}
