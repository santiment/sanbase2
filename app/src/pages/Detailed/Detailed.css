.detailed {
  min-width: 100%;
}

.panel-chart {
  padding-top: 0;
}

.datailed-project-description {
  font-size: 16px;
}

.detailed-buttons {
  width: 50%;
  height: 100%;
  position: relative;
}

.detailed-buttons button.add-to-dashboard {
  border: none;
  border-radius: 4px;
  padding: 8px 11px;
  text-transform: uppercase;
  letter-spacing: 1px;
  background-color: #009663;
  color: #fff;
  font-size: 12px;
  font-weight: 400;
  opacity: 0.7;
  position: absolute;
  right: 0;
  bottom: 0;
}

.detailed-buttons button.add-to-dashboard:hover {
  opacity: 1;
  background-color: #025aa5;
  border-color: #01549b;
}

.detailed .nav {
  border-bottom: 1px solid #ccc;
  display: flex;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
}

.detailed .nav li {
  display: list-item;
  text-align: -webkit-match-parent;
  margin-bottom: -1px;
}

.detailed .nav li .nav-link {
  border: 1px solid;
  border-color: #ccc rgba(0, 0, 0, 0) #ccc #ccc;
  background-color: #f2f2f2;
  color: #3d4051;
  display: block;
}

.detailed .main-chart .nav li {
  width: 50%;
}

.detailed .activity-panel .nav .nav-link {
  font-size: 1.25rem;
  font-weight: 500;
  color: #3d4051;
  padding: 3px 8px;
}

.detailed .main-chart .nav .nav-link {
  font-size: 24px;
  font-weight: 300;
  color: #171717;
  padding: 4px 15px;
}

.detailed .nav li:last-child .nav-link {
  border-color: #ccc;
}

.detailed .activity-panel .nav li {
  width: 20%;
}

.detailed .nav li.active .nav-link {
  background-color: #fff;
  border-color: #ccc rgba(0, 0, 0, 0) #fff #ccc;
}

.detailed .nav li.active:last-child .nav-link {
  background-color: #fff;
  border-color: #ccc #ccc #fff #ccc;
}

.detailed .analysis {
  display: flex;
  margin: 16px;
  justify-content: space-between;
}

.detailed .analysis .panel {
  width: 33%;
  margin: 0;
}

.row-info {
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  padding: 6px 0;
}

.night-mode .row-info {
  border-bottom-color: #2d3239;
}

.row-info .value {
  text-align: right;
}

.detailed .information {
  display: flex;
  margin: 0;
  margin-bottom: 16px;
  justify-content: space-between;
  flex-direction: column;
}

.detailed .information .panel {
  flex: 1 1 0;
  margin: 0;
  margin-top: 16px;
}

.detailed .top-eth-transaction {
  display: flex;
  justify-content: space-between;
}

.detailed .top-eth-transaction__hash {
  text-overflow: ellipsis;
  overflow: hidden;
  width: 70%;
  height: 2em;
  white-space: nowrap;
}

@media (max-width: 768px) {
  .App {
    background-color: white;
  }

  .panel {
    box-shadow: none;
    padding: 0;
  }
}

@media (min-width: 1024px) {
  .detailed-head {
    margin: 0 16px;
    margin-bottom: 1em;
    flex-direction: row;
  }

  .detailed-name h1 {
    font-size: 1.8em;
    letter-spacing: 0.5px;
  }

  .detailed-price {
    line-height: 1.4;
    margin-top: auto;
  }

  .detailed-price-usd {
    justify-content: flex-start;
    font-size: 0.8em;
  }

  .detailed-price-description {
    text-align: left;
    font-size: 0.5em;
    font-weight: 300;
    margin-bottom: 0;
  }

  .detailed .information {
    flex-direction: row;
    margin-right: 16px;
  }

  .detailed .information .panel {
    margin-top: 0;
    margin-right: 0;
    margin-left: 16px;
  }
}
