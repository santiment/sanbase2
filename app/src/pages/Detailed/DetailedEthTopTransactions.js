import React, { Fragment } from 'react'
import { Label } from 'semantic-ui-react'
import PanelBlock from './../../components/PanelBlock'
import moment from 'moment'
import { formatNumber } from './../../utils/formatting'
import ReactTable from 'react-table'
import SmoothDropdown from '../../components/SmoothDropdown/SmoothDropdown'
import SmoothDropdownItem from '../../components/SmoothDropdown/SmoothDropdownItem'
import './DetailedEthTopTransactions.css'

const getAddressMarkup = ({ address, isExchange }) => (
  <Fragment>
    {isExchange ? <Label color='yellow'>exchange</Label> : null}
    <a href={`https://etherscan.io/address/${address}`}>{address}</a>
  </Fragment>
)

const DetailedEthTopTransactionsAddressCell = ({ value }) => (
  <SmoothDropdownItem
    showIf={({ currentTarget: trigger }) =>
      trigger.offsetWidth + 10 >= trigger.parentNode.offsetWidth
    }
    trigger={getAddressMarkup(value)}
  >
    {value.address}
  </SmoothDropdownItem>
)

const COLUMNS = [
  {
    Header: 'Time',
    accessor: 'datetime',
    sortMethod: (a, b) => {
      return moment(a).isAfter(moment(b)) ? 1 : -1
    }
  },
  {
    Header: 'Value',
    accessor: 'trxValue',
    sortMethod: (a, b) => {
      return parseFloat(a) > parseFloat(b) ? 1 : -1
    }
  },
  {
    Header: 'From',
    accessor: 'fromAddress',
    Cell: DetailedEthTopTransactionsAddressCell,
    sortable: false
  },
  {
    Header: 'To',
    accessor: 'toAddress',
    Cell: DetailedEthTopTransactionsAddressCell,
    sortable: false
  }
]

const DetailedEthTopTransactions = ({ Project }) => {
  const DATA = Project.project.tokenTopTransactions
    .slice(0, 10)
    .map(({ trxValue, fromAddress, toAddress, datetime }) => ({
      trxValue: formatNumber(trxValue),
      fromAddress: fromAddress,
      toAddress: toAddress,
      datetime: moment(datetime).format('YYYY-MM-DD HH:mm:ss')
    }))
  return (
    <PanelBlock isLoading={Project.loading} title='Top ETH Transactions'>
      <SmoothDropdown verticalMotion>
        <ReactTable
          data={DATA}
          columns={COLUMNS}
          showPagination={false}
          minRows={2}
          className='DetailedEthTopTransactions'
        />
      </SmoothDropdown>
    </PanelBlock>
  )
}

export default DetailedEthTopTransactions
