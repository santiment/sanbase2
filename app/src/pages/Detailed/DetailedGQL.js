import gql from 'graphql-tag'

export const projectGQL = gql`
  query projectGQL($id: ID!) {
    project(
      id: $id,
    ){
      id,
      name,
      ticker,
      description,
      websiteLink,
      facebookLink,
      githubLink,
      redditLink,
      twitterLink,
      whitepaperLink,
      slackLink,
      btcBalance,
      projectTransparency,
      projectTransparencyDescription,
      projectTransparencyStatus,
      tokenAddress,
      fundsRaisedIcos { amount, currencyCode },
      roiUsd,
      priceUsd,
      priceBtc,
      volumeUsd,
      ethBalance,
      ethSpent,
      marketcapUsd,
      tokenDecimals,
      rank,
      totalSupply,
      percentChange24h,
    }
  }
`
export const queryTwitterHistory = gql`
  query queryTwitterHistory($ticker:String, $from: DateTime, $to: DateTime, $interval: String) {
    historyTwitterData(
      ticker: $ticker,
      from: $from,
      to: $to,
      interval: $interval
    ) {
      datetime
      followersCount
      __typename
    }
  }
`

export const queryTwitterData = gql`
  query queryTwitterData($ticker:String) {
    twitterData(ticker: $ticker) {
      datetime
      followersCount
      twitterName
    }
  }
`

export const queryHistoryPrice = gql`
  query queryHistoryPrice($ticker: String, $from: DateTime, $to: DateTime, $interval: String) {
    historyPrice(
      ticker: $ticker,
      from: $from,
      to: $to,
      interval: $interval
    ) {
      priceBtc,
      priceUsd,
      volume,
      datetime,
      marketcap
    }
}`

export const queryGithubActivity = gql`
  query queryGithubActivity($ticker: String, $from: DateTime, $to: DateTime, $interval: String, $transform: String, $movingAverageInterval: Int) {
    githubActivity(
      ticker: $ticker,
      from: $from,
      to: $to,
      interval: $interval,
      transform: $transform,
      movingAverageInterval: $movingAverageInterval
    ) {
      datetime,
      activity
    }
}`

export const queryBurnRate = gql`
  query queryBurnRate($ticker:String, $from: DateTime, $to: DateTime) {
    burnRate(
      ticker: $ticker,
      from: $from,
      to: $to
    ) {
      datetime
      burnRate
      __typename
    }
}`

export const queryTransactionVolume = gql`
  query queryTransactionVolume($ticker:String, $from: DateTime, $to: DateTime) {
    transactionVolume(
      ticker: $ticker,
      from: $from,
      to: $to
    ) {
      datetime
      transactionVolume
      __typename
    }
}`
