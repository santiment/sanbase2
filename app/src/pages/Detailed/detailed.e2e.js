/* eslint-env jasmine */
import puppeteer from 'puppeteer'

let browser = null
let page = null
let mobilePage = null

beforeAll(async() => {
  jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000
  const launchOptions = process.env.CI
    ? {}
    : {headless: true, slowMo: 5, ignoreHTTPSErrors: true}

  // Workaround till https://github.com/GoogleChrome/puppeteer/issues/290 is fixed
  // if (process.env.LAUNCH_CHROME_NO_SANDBOX) {
    // console.warn('Launching Chrome with "--no-sandbox" option. ' +
      // 'This is not recommended due to security reasons!')
  Object.assign(launchOptions, { args: ['--no-sandbox', '--disable-setuid-sandbox'] })
  // }

  browser = await puppeteer.launch(launchOptions)
  page = await browser.newPage()
  await page.setViewport({
    width: 1366,
    height: 768
  })
  mobilePage = await browser.newPage()
  await mobilePage.setViewport({
    width: 360,
    height: 640,
    isMobile: true
  })
})

afterAll(async () => {
  await browser.close()
})

const detailedPageTests = () => {
  it('it should loads correctly', async () => {
    await page.goto('http://localhost:3000/projects/ppt')
    await mobilePage.goto('http://localhost:3000/projects/ppt')
    await page.waitForSelector('.datailed-project-description')
    await mobilePage.waitForSelector('.datailed-project-description')
    await page.screenshot({path: '.screenshots/ppt-detailed-page-desktop.png'})
    await mobilePage.screenshot({path: '.screenshots/ppt-detailed-page-mobile.png'})
  })

  it('Search should works correctly', async () => {
    await page.goto('http://localhost:3000/projects/eos')
    await page.waitForSelector('.search-data-loaded')
    await page.focus('#search-input')
    await page.keyboard.type('san')
    await page.waitForSelector('#search-result')
    const searchResult = await page.evaluate(() => {
      const resultDiv = document.querySelector('#search-result')
      return resultDiv.innerText || null
    })
    expect(searchResult).toBe('Santiment Network Token (SAN)')
    await page.screenshot({path: '.screenshots/eos-detailed-page-desktop.png'})
    await page.keyboard.press('Enter')
    await page.waitFor(4000)
    await page.screenshot({path: '.screenshots/san-detailed-page-desktop.png'})
  })
}

if (process.env.CI === true) {
  describe.skip('Detailed page', detailedPageTests)
} else {
  describe('Detailed page', detailedPageTests)
}
