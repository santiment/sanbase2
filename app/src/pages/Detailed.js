import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import {
  compose,
  pure,
  lifecycle
} from 'recompose'
import { Redirect } from 'react-router-dom'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import ProjectIcon from './../components/ProjectIcon'
import PanelBlock from './../components/PanelBlock'
import { retrieveProjects } from './Cashflow.actions.js'
import './Detailed.css'

const propTypes = {
  match: PropTypes.object.isRequired,
  projects: PropTypes.array.isRequired,
  loading: PropTypes.bool.isRequired
}

export const Detailed = ({match, projects, loading}) => {
  if (loading) {
    return (
      <div className='page detailed'>
        <h2>Loading...</h2>
      </div>
    )
  }
  const selectedTicker = match.params.ticker
  const project = projects.find(el => {
    const ticker = el.ticker || ''
    return ticker.toLowerCase() === selectedTicker
  })
  if (!project) {
    return (
      <Redirect to={{
        pathname: '/'
      }} />
    )
  }
  return (
    <div className='page detailed'>
      <div className='detailed-head'>
        <div className='detailed-name'>
          <h1><ProjectIcon name={project.name} size={28} /> {project.name} ({project.ticker.toUpperCase()})</h1>
          <p>Manage entire organisations using the blockchain.</p>
        </div>
        <div className='detailed-buttons'>
          <a href='#' className='add-to-dashboard'>
            <i className='fa fa-plus' />
            &nbsp; Add to Dashboard
          </a>
        </div>
      </div>
      <div className='panel'>
        <Tabs className='main-chart'>
          <TabList className='nav'>
            <Tab className='nav-item' selectedClassName='active'>
              <a className='nav-link' href='#'>
                $2.29 USD (not real) &nbsp;
                <span className='diff down'>
                  <i className='fa fa-caret-down' />
                    &nbsp; 8.87% (not real)
                </span>
              </a>
            </Tab>
            <Tab className='nav-item' selectedClassName='active'>
              <a className='nav-link' href='#'>
                2.29 BTC (not real) &nbsp;
                <span className='diff up'>
                  <i className='fa fa-caret-up' />
                    &nbsp; 8.87% (not real)
                </span>
              </a>
            </Tab>
          </TabList>
          <TabPanel>
            1
          </TabPanel>
          <TabPanel>
            2
          </TabPanel>
        </Tabs>
      </div>
      <div className='panel'>
        <Tabs className='activity-panel'>
          <TabList className='nav'>
            <Tab className='nav-item' selectedClassName='active'>
              <a className='nav-link' href='#'>
                Social Mentions
              </a>
            </Tab>
            <Tab className='nav-item' selectedClassName='active'>
              <a className='nav-link' href='#'>
                Social Activity over Time
              </a>
            </Tab>
            <Tab className='nav-item' selectedClassName='active'>
              <a className='nav-link' href='#'>
                Sentiment/Intensity
              </a>
            </Tab>
            <Tab className='nav-item' selectedClassName='active'>
              <a className='nav-link' href='#'>
                Github Activity
              </a>
            </Tab>
            <Tab className='nav-item' selectedClassName='active'>
              <a className='nav-link' href='#'>
                SAN Community
              </a>
            </Tab>
          </TabList>
          <TabPanel>
            Social Mentions
          </TabPanel>
          <TabPanel>
            Social Activity over Time
          </TabPanel>
          <TabPanel>
            Sentiment/Intensity
          </TabPanel>
          <TabPanel>
            Github Activity
          </TabPanel>
          <TabPanel>
            SAN Community
          </TabPanel>
        </Tabs>
      </div>
      <PanelBlock title='Blockchain Analytics' />
      <div className='row analysis'>
        <PanelBlock title='Signals/Volatility' />
        <PanelBlock title='Expert Analyses' />
        <PanelBlock title='News/Press' />
      </div>
      <div className='row information'>
        <PanelBlock title='General Info' />
        <PanelBlock title='Financials' />
      </div>
    </div>
  )
}

Detailed.propTypes = propTypes

const mapStateToProps = state => {
  return {
    projects: state.projects.items,
    loading: state.projects.isLoading
  }
}

const mapDispatchToProps = dispatch => {
  return {
    // TODO: have to retrive only selected project
    retrieveProjects: () => dispatch(retrieveProjects)
  }
}

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  lifecycle({
    componentDidMount () {
      this.props.retrieveProjects()
    }
  }),
  pure
)

export default enhance(Detailed)
