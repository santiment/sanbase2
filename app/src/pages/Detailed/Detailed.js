import React from 'react'
import PropTypes from 'prop-types'
import {
  compose,
  lifecycle,
  withState
} from 'recompose'
import moment from 'moment'
import { Redirect } from 'react-router-dom'
import { Helmet } from 'react-helmet'
import { graphql, withApollo } from 'react-apollo'
import PanelBlock from './../../components/PanelBlock'
import GeneralInfoBlock from './GeneralInfoBlock'
import FinancialsBlock from './FinancialsBlock'
import ProjectChartContainer from './../../components/ProjectChart/ProjectChartContainer'
import Panel from './../../components/Panel'
import Search from './../../components/SearchContainer'
import { calculateBTCVolume, calculateBTCMarketcap } from '../../utils/utils'
import allProjectsGQL from './../Projects/allProjectsGQL'
import DetailedHeader from './DetailedHeader'
import {
  projectGQL,
  queryTwitterData,
  queryTwitterHistory,
  queryHistoryPrice,
  queryBurnRate,
  queryGithubActivity,
  queryTransactionVolume
} from './DetailedGQL'
import './Detailed.css'

const propTypes = {
  match: PropTypes.object.isRequired
}

const getProjectIDByTicker = (match, allProjects = null) => {
  const selectedTicker = match.params.ticker
  const project = (allProjects || []).find(el => {
    const ticker = el.ticker || ''
    return ticker.toLowerCase() === selectedTicker
  })
  return parseInt((project || {}).id, 10) || null
}

export const Detailed = ({
  match,
  history,
  location,
  TwitterData = {
    loading: true,
    error: false,
    twitterData: []
  },
  TwitterHistoryData = {
    loading: true,
    error: false,
    twitterHistoryData: []
  },
  HistoryPrice = {
    loading: true,
    error: false,
    historyPrice: []
  },
  GithubActivity = {
    loading: true,
    error: false,
    githubActivity: []
  },
  BurnRate = {
    loading: true,
    error: false,
    burnRate: []
  },
  TransactionVolume = {
    loading: true,
    error: false,
    transactionVolume: []
  },
  Project = {
    project: undefined,
    loading: true,
    error: false
  },
  changeChartVars,
  isDesktop,
  projectId = -1,
  projects = []
}) => {
  const project = Project.project
  if (!projectId) {
    return (
      <Redirect to={{
        pathname: '/'
      }} />
    )
  }

  const twitter = {
    history: {
      error: TwitterHistoryData.error || false,
      loading: TwitterHistoryData.loading,
      items: TwitterHistoryData.historyTwitterData || []
    },
    data: {
      error: TwitterData.error || false,
      loading: TwitterData.loading,
      followersCount: TwitterData.twitterData
        ? TwitterData.twitterData.followersCount
        : undefined
    }
  }

  const price = {
    history: {
      loading: HistoryPrice.loading,
      items: HistoryPrice.historyPrice
        ? HistoryPrice.historyPrice.map(item => {
          const volumeBTC = calculateBTCVolume(item)
          const marketcapBTC = calculateBTCMarketcap(item)
          return {...item, volumeBTC, marketcapBTC}
        })
        : []
    }
  }

  const github = {
    history: {
      loading: GithubActivity.loading,
      items: GithubActivity.githubActivity || []
    }
  }

  const burnRate = {
    ...BurnRate,
    loading: BurnRate.loading,
    error: BurnRate.error || false,
    items: BurnRate.burnRate || []
  }

  const transactionVolume = {
    loading: TransactionVolume.loading,
    error: TransactionVolume.error || false,
    items: TransactionVolume.transactionVolume || []
  }

  const projectContainerChart = project &&
    <ProjectChartContainer
      routerHistory={history}
      location={location}
      isDesktop={isDesktop}
      twitter={twitter}
      price={price}
      github={github}
      burnRate={burnRate}
      tokenDecimals={Project.project ? Project.project.tokenDecimals : undefined}
      transactionVolume={transactionVolume}
      onDatesChange={(from, to, interval, ticker) => {
        changeChartVars({
          from,
          to,
          interval,
          ticker
        })
      }}
      ticker={project.ticker} />

  return (
    <div className='page detailed'>
      <Helmet>
        <title>{Project.loading
          ? 'SANbase...'
          : `${Project.project.ticker} project page`}
        </title>
      </Helmet>
      {!isDesktop && <Search />}
      <DetailedHeader {...Project} />
      {isDesktop
        ? <Panel zero>{projectContainerChart}</Panel>
        : <div>{projectContainerChart}</div>}
      <div className='information'>
        <PanelBlock
          isLoading={Project.loading}
          title='General Info'>
          <GeneralInfoBlock {...Project.project} />
        </PanelBlock>
        <PanelBlock
          isLoading={Project.loading}
          title='Financials'>
          <FinancialsBlock {...Project.project} />
        </PanelBlock>
      </div>
    </div>
  )
}

Detailed.propTypes = propTypes

const enhance = compose(
  withApollo,
  withState('chartVars', 'changeChartVars', {
    from: undefined,
    to: undefined,
    interval: undefined,
    ticker: undefined
  }),
  withState('projectId', 'changeProjectId', undefined),
  withState('projects', 'changeProjects', []),
  lifecycle({
    componentDidMount () {
      this.props.client.query({
        query: allProjectsGQL
      }).then(response => {
        const id = getProjectIDByTicker(this.props.match, response.data.allProjects)
        this.props.changeProjectId(id)
        this.props.changeProjects(response.data.allProjects)
      })
    },
    componentDidUpdate (prevProps, prevState) {
      if (this.props.match.params.ticker !== prevProps.match.params.ticker &&
        this.props.projects.length > 0) {
        const id = getProjectIDByTicker(this.props.match, this.props.projects)
        this.props.changeProjectId(id)
      }
    }
  }),
  graphql(projectGQL, {
    name: 'Project',
    props: ({Project}) => ({
      Project: {
        loading: Project.loading,
        empty: !Project.hasOwnProperty('project'),
        error: Project.error,
        errorMessage: Project.error ? Project.error.message : '',
        project: Project.project
      }
    }),
    options: ({projectId}) => ({
      skip: !projectId,
      errorPolicy: 'all',
      variables: {
        id: projectId
      }
    })
  }),
  graphql(queryTwitterData, {
    name: 'TwitterData',
    options: ({chartVars}) => {
      const { ticker } = chartVars
      return {
        skip: !ticker,
        errorPolicy: 'all',
        variables: {
          ticker
        }
      }
    }
  }),
  graphql(queryTwitterHistory, {
    name: 'TwitterHistoryData',
    options: ({chartVars}) => {
      const {from, to, ticker} = chartVars
      return {
        skip: !from,
        errorPolicy: 'all',
        variables: {
          from,
          to,
          ticker
        }
      }
    }
  }),
  graphql(queryHistoryPrice, {
    name: 'HistoryPrice',
    options: ({chartVars}) => {
      const {from, to, ticker, interval} = chartVars
      return {
        skip: !from,
        errorPolicy: 'all',
        variables: {
          from,
          to,
          ticker,
          interval
        }
      }
    }
  }),
  graphql(queryGithubActivity, {
    name: 'GithubActivity',
    options: ({match, chartVars}) => {
      const {from, to, ticker} = chartVars
      return {
        skip: !from,
        variables: {
          from: from ? moment(from).subtract(7, 'days') : undefined,
          to,
          ticker,
          interval: '1d',
          transform: 'movingAverage',
          movingAverageInterval: 7
        }
      }
    }
  }),
  graphql(queryBurnRate, {
    name: 'BurnRate',
    options: ({chartVars}) => {
      const {from, to, ticker} = chartVars
      return {
        skip: !from,
        errorPolicy: 'all',
        variables: {
          from,
          to,
          ticker
        }
      }
    }
  }),
  graphql(queryTransactionVolume, {
    name: 'TransactionVolume',
    options: ({chartVars}) => {
      const {from, to, ticker} = chartVars
      return {
        skip: !from,
        errorPolicy: 'all',
        variables: {
          from,
          to,
          ticker
        }
      }
    }
  })
)

export default enhance(Detailed)
