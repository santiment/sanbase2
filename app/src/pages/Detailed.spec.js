/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Detailed } from './Detailed'

const projects = [
  {
    'balance': '41994.09',
    'id': 12,
    'logo_url': 'aeternity.png',
    'market_cap_usd': '133759577.0',
    'name': 'Aeternity',
    'ticker': 'AE',
    'wallets': [
      {
        'address': '0x15c19E6c203E2c34D1EDFb26626bfc4F65eF96F0',
        'balance': '41994.09',
        'last_outgoing': null,
        'tx_out': null
      }
    ]
  },
  {
    'balance': '264617.29',
    'id': 15,
    'logo_url': 'aragon.png',
    'market_cap_usd': '66649211.0',
    'name': 'Aragon',
    'ticker': 'ANT',
    'wallets': [
      {
        'address': '0xcafE1A77e84698c83CA8931F54A755176eF75f2C',
        'balance': '264617.29',
        'last_outgoing': null,
        'tx_out': null
      }
    ]
  }
]

describe('Project detail page container', () => {
  it('it should render correctly', () => {
    const match = {
      params: {ticker: 'ae'}
    }
    const pdp = shallow(<Detailed
      projects={projects}
      loading={false}
      price={{
        loading: true
      }}
      match={match}
      />)
    expect(toJson(pdp)).toMatchSnapshot()
  })

  it('it should redirect to /', () => {
    const match = {
      params: {ticker: 'whatsaaa'}
    }
    const pdp = shallow(<Detailed
      projects={projects}
      price={{
        loading: true
      }}
      loading={false}
      match={match}
      />)
    expect(toJson(pdp)).toMatchSnapshot()
  })
})
