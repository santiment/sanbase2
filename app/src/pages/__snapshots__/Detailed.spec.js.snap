// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project detail page container it should redirect to / 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/",
    }
  }
/>
`;

exports[`Project detail page container it should render correctly 1`] = `
<div
  className="page detailed"
>
  <div
    className="detailed-head"
  >
    <div
      className="detailed-name"
    >
      <h1>
        <ProjectIcon
          name="Aeternity"
          size={28}
        />
         
        Aeternity
         (
        AE
        )
      </h1>
      <p>
        Manage entire organisations using the blockchain.
      </p>
    </div>
    <div
      className="detailed-buttons"
    >
      <a
        className="add-to-dashboard"
        href="#"
      >
        <i
          className="fa fa-plus"
        />
          Add to Dashboard
      </a>
    </div>
  </div>
  <div
    className="panel"
  >
    <Tabs
      className="main-chart"
      defaultFocus={false}
      defaultIndex={null}
      forceRenderTabPanel={false}
      selectedIndex={null}
    >
      <TabList
        className="nav"
      >
        <Tab
          className="nav-item"
          disabledClassName="react-tabs__tab--disabled"
          focus={false}
          id={null}
          panelId={null}
          selected={false}
          selectedClassName="active"
        >
          <a
            className="nav-link"
            href="#"
          >
            $2.29 USD (not real)  
            <span
              className="diff down"
            >
              <i
                className="fa fa-caret-down"
              />
                8.87% (not real)
            </span>
          </a>
        </Tab>
        <Tab
          className="nav-item"
          disabledClassName="react-tabs__tab--disabled"
          focus={false}
          id={null}
          panelId={null}
          selected={false}
          selectedClassName="active"
        >
          <a
            className="nav-link"
            href="#"
          >
            2.29 BTC (not real)  
            <span
              className="diff up"
            >
              <i
                className="fa fa-caret-up"
              />
                8.87% (not real)
            </span>
          </a>
        </Tab>
      </TabList>
      <TabPanel
        className="react-tabs__tab-panel"
        forceRender={false}
        selectedClassName="react-tabs__tab-panel--selected"
      >
        1
      </TabPanel>
      <TabPanel
        className="react-tabs__tab-panel"
        forceRender={false}
        selectedClassName="react-tabs__tab-panel--selected"
      >
        2
      </TabPanel>
    </Tabs>
  </div>
  <div
    className="panel"
  >
    <Tabs
      className="activity-panel"
      defaultFocus={false}
      defaultIndex={null}
      forceRenderTabPanel={false}
      selectedIndex={null}
    >
      <TabList
        className="nav"
      >
        <Tab
          className="nav-item"
          disabledClassName="react-tabs__tab--disabled"
          focus={false}
          id={null}
          panelId={null}
          selected={false}
          selectedClassName="active"
        >
          <a
            className="nav-link"
            href="#"
          >
            Social Mentions
          </a>
        </Tab>
        <Tab
          className="nav-item"
          disabledClassName="react-tabs__tab--disabled"
          focus={false}
          id={null}
          panelId={null}
          selected={false}
          selectedClassName="active"
        >
          <a
            className="nav-link"
            href="#"
          >
            Social Activity over Time
          </a>
        </Tab>
        <Tab
          className="nav-item"
          disabledClassName="react-tabs__tab--disabled"
          focus={false}
          id={null}
          panelId={null}
          selected={false}
          selectedClassName="active"
        >
          <a
            className="nav-link"
            href="#"
          >
            Sentiment/Intensity
          </a>
        </Tab>
        <Tab
          className="nav-item"
          disabledClassName="react-tabs__tab--disabled"
          focus={false}
          id={null}
          panelId={null}
          selected={false}
          selectedClassName="active"
        >
          <a
            className="nav-link"
            href="#"
          >
            Github Activity
          </a>
        </Tab>
        <Tab
          className="nav-item"
          disabledClassName="react-tabs__tab--disabled"
          focus={false}
          id={null}
          panelId={null}
          selected={false}
          selectedClassName="active"
        >
          <a
            className="nav-link"
            href="#"
          >
            SAN Community
          </a>
        </Tab>
      </TabList>
      <TabPanel
        className="react-tabs__tab-panel"
        forceRender={false}
        selectedClassName="react-tabs__tab-panel--selected"
      >
        Social Mentions
      </TabPanel>
      <TabPanel
        className="react-tabs__tab-panel"
        forceRender={false}
        selectedClassName="react-tabs__tab-panel--selected"
      >
        Social Activity over Time
      </TabPanel>
      <TabPanel
        className="react-tabs__tab-panel"
        forceRender={false}
        selectedClassName="react-tabs__tab-panel--selected"
      >
        Sentiment/Intensity
      </TabPanel>
      <TabPanel
        className="react-tabs__tab-panel"
        forceRender={false}
        selectedClassName="react-tabs__tab-panel--selected"
      >
        Github Activity
      </TabPanel>
      <TabPanel
        className="react-tabs__tab-panel"
        forceRender={false}
        selectedClassName="react-tabs__tab-panel--selected"
      >
        SAN Community
      </TabPanel>
    </Tabs>
  </div>
  <PanelBlock
    classes=""
    content=""
    title="Blockchain Analytics"
  />
  <div
    className="row analysis"
  >
    <PanelBlock
      classes=""
      content=""
      title="Signals/Volatility"
    />
    <PanelBlock
      classes=""
      content=""
      title="Expert Analyses"
    />
    <PanelBlock
      classes=""
      content=""
      title="News/Press"
    />
  </div>
  <div
    className="row information"
  >
    <PanelBlock
      classes=""
      content=""
      title="General Info"
    />
    <PanelBlock
      classes=""
      content=""
      title="Financials"
    />
  </div>
</div>
`;
